src/plugins/action/action_impl.cpp

```diff
@@ -234,31 +234,31 @@ void ActionImpl::kill_async(const Action::ResultCallback& callback) const
     });
 }
 
-void ActionImpl::reboot_async(const Action::ResultCallback& callback) const
+void ActionImpl::reboot_async(const Action::ResultCallback& callback, const uint8_t component) const
```

Comment:
Shouldn't it be more something like an enum? e.g. `reboot(GIMBAL)`, `reboot(AUTOPILOT)`, `reboot(ALL)`?

Reply:
Yes, it could be an enum or the MAVLink component ID, I'd say.

Reply:
Oh, right. There could be 2 cameras or 2 gimbals, again the same thing. But then shouldn't it be `drone.gimbal.reboot()` or `drone.camera.reboot()`? And in the future, when we support multiple cameras, it becomes something like `drone.camera[0].reboot()`.

Reply:
True, that would be nicer. However, @shakeebbb wants to reboot only the autopilot.

Reply:
Shouldn't MAV_COMPONENT enum be enough to handle multiple cameras and gimbals on the mavlink network?

Reply:
I see. What about `drone.action.reboot(bool autopilot_only)`? So that the action plugin can reboot only the autopilot OR everything. And then for individual components, this goes into their corresponding plugin. Would that work?